### PR TITLE
fix: added migrations for old input widget

### LINKED ISSUE #6809: [Bug]The Application with Input widget breaks as the Valid toggle is not enabled  by default for the  existing application
## Description

[What happened]The Application with Input widget breaks as the Valid toggle is not enabled by default for the  existing application

### Steps to reproduce the behaviour:

1. Go to canvas
2. Generate a new app for S3 datasource 
3. Now try to search a file name from the input box
4. The error is displayed when the user enters the text

This would be an issue for the previously developed apps because the Valid toggle will be in a disabled state for the existing apps
image

### Important Details

- Version: [Cloud ]
- OS: [MacOS]
- Browser [chrome]
- Environment [ release]

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> add migration for input validation

Fixes #6809 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ]  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ]  I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: bug/inputWidget-valid-migration 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.04 **(0.02)** | 36.92 **(0.03)** | 33.98 **(0.03)** | 55.59 **(0.02)**
 :green_circle: | app/client/src/utils/WidgetPropsUtils.tsx | 75.59 **(0.71)** | 66.67 **(0.2)** | 71.13 **(1.24)** | 74.67 **(0.76)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 37.72 **(-0.88)** | 36.21 **(0)** | 55.08 **(-0.27)**</details>

### CODE REVIEW

**Review by marks0351 — DISMISSED** [review]

**Comment by somangshu** [comment]
@techbhavin #6857 should also be fixed here, Can you please link it

**Comment by somangshu** [comment]
> @techbhavin #6857 should also be fixed here, Can you please link it

After discussion the above will be taken up seperately

File: app/client/src/utils/WidgetPropsUtils.tsx [hunk 1]
```diff
@@ -897,6 +902,25 @@ export const migrateToNewMultiSelect = (
   }
   return currentDSL;
 };
+
+const migrateButtonWidgetValidation = (
+  currentDSL: ContainerWidgetProps<WidgetProps>,
+) => {
+  if (currentDSL.type === WidgetTypes.INPUT_WIDGET) {
+    if (!has(currentDSL, "validation")) {
+      currentDSL.validation = true;
```
Comment (jsartisan) [inline app/client/src/utils/WidgetPropsUtils.tsx:911]: not sure how this line fixes this issue. other than this, everything looks good.

**Review by jsartisan — APPROVED** [review]

**Comment by jsartisan** [comment]
@somangshu @techbhavin the code-wise LGTM. just left a small comment.

**Comment by jsartisan** [comment]
Also, there is one small merge conflict.

**Comment by somangshu** [comment]
@marks0351 can you please review this at priority, 

@ramsaptami @RakshaKShetty we have to verify the issue here as soon as possible, Gotta close this before the release is frozen

**Review by marks0351 — APPROVED** [review]

**Comment by ramsaptami** [comment]
@somangshu Verified and tested ok on DP

**Comment by somangshu** [comment]
@techbhavin please merge the latest release here

**Review by somangshu — APPROVED** [review]

**Pull request merged**